### PR TITLE
Adds osu! commands

### PR DESCRIPTION
**Pull request by pedrofracassi**
### Commands
- [X] `osuprofile <mode> <nickname>`
- [ ] `osubest <mode> <nickname>` **WIP** THIS ONE WILL BE SLOW AS FFFF....
- [ ] `osubeatmap <id>`
- [ ] `osurecent <mode> <nickname>`

### OsuUtils
- [X] `modes` - Valid modes list
- [ ] `calculateAccuracy` - Calculates the accuracy based on mode, hits and beatmap. ([See this for reference](https://osu.ppy.sh/help/wiki/Accuracy?locale=en))
- [ ] `calculateMods` - Takes a number and returns an array of mods used in a certain play ([See this for reference](https://github.com/ppy/osu-api/wiki#mods))

### Environment variables
- `OSU_API_KEY` - Can be obtained [here](https://osu.ppy.sh/p/api)

### CODE REVIEW

File: src/commands/game/osu/osuprofile.js [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+  }
+
+  canLoad () {
+    return !!this.client.apis.osu
+  }
+
+  async run (message, args) {
+    const embed = new SwitchbladeEmbed(message.author)
+    if (args.length >= 2) { // Check if enough args were passed
+      if (modes.indexOf(args[0]) > -1) { // Check if the first argument is a valid gamemode
```
Comment (davipatury) [inline src/commands/game/osu/osuprofile.js]: you can use `modes.includes(args[0])`

**Comment by pedrofracassi** [comment]
ha this is going to be fun to fix

**Comment by LBlend** [comment]
osu! map PP calculation. Calculate the maximum obtainable PP for a map

Example of a tool that does this:
https://osu.bitti.io/

**Comment by pedrofracassi** [comment]
@LBlend woah, it requires you to upload the `.osu` file?

**Comment by pedrofracassi** [comment]
oooo wait
image

this is good

**Comment by pedrofracassi** [comment]
I should proooobably rewrite this, shouldn't I?

**Comment by bolsomito** [comment]
yes

**Comment by Doges** [comment]
pedro is remaking this l8

**Pull request closed without merging**